packages/wonder-blocks-link/components/link.test.js

```diff
@@ -1,12 +1,58 @@
 // @flow
 import React from "react";
-import {mount} from "enzyme";
+import {MemoryRouter, Route, Switch} from "react-router-dom";
 
+import {mount, unmountAll} from "../../../utils/testing/mount.js";
 import Link from "./link.js";
 
 describe("Link", () => {
-    test("TODO", () => {
-        const wrapper = mount(<Link href="#">Hello World!</Link>);
-        wrapper.simulate("click");
+    beforeEach(() => {
+        unmountAll();
     });
-});
\ No newline at end of file
+
+    test("client-side navigation", () => {
+        const wrapper = mount(
+            <MemoryRouter>
+                <div>
+                    <Link testId="link" href="/foo" clientNav={true}>
+                        Click me!
+                    </Link>
+                    <Switch>
+                        <Route path="/foo">
+                            <div id="foo">Hello, world!</div>
+                        </Route>
+                    </Switch>
+                </div>
+            </MemoryRouter>,
+        );
+
+        expect(wrapper.find("#foo").exists()).toBe(false);
+        const buttonWrapper = wrapper.find(`[data-test-id="link"]`).first();
+        expect(buttonWrapper.exists()).toBe(true);
+        buttonWrapper.simulate("click", {button: 0});
+        expect(wrapper.find("#foo").exists()).toBe(true);
+    });
+
+    test("client-side navigation without 'clientNav' prop fails", () => {
+        const wrapper = mount(
+            <MemoryRouter>
+                <div>
+                    <Link testId="link" href="/foo">
+                        Click me!
+                    </Link>
+                    <Switch>
+                        <Route path="/foo">
+                            <div id="foo">Hello, world!</div>
+                        </Route>
+                    </Switch>
+                </div>
+            </MemoryRouter>,
+        );
+
+        expect(wrapper.find("#foo").exists()).toBe(false);
+        const buttonWrapper = wrapper.find(`[data-test-id="link"]`).first();
+        expect(buttonWrapper.exists()).toBe(true);
+        buttonWrapper.simulate("click", {button: 0});
+        expect(wrapper.find("#foo").exists()).toBe(false);
+    });
```

Comment:
I'd love to see `// Arrange`, `// Act`, `// Assert` structure in these tests to make it clear what the test is doing.